src/aind_codeocean_utils/codeocean_job.py

```diff
@@ -0,0 +1,327 @@
+
+    #     return data_asset_reg_response
+
+    def capture_result(
+        self,
+        computation_id: str,
+        asset_name: str,
+        mount: str,
+        tags: List[str],
+        custom_metadata: Optional[dict] = None,
+        viewable_to_everyone: bool = False,
+    ):
+        """
+        Capture a result as a data asset. Can also share it with everyone.
+        Parameters
+        ----------
+        computation_id : str
+          ID of the computation
+        asset_name : str
+          Name to give the data asset
+        mount : str
+          Mount folder name for the data asset.
+        tags : List[str]
+          List of tags to describe the data asset.
+        custom_metadata : Optional[dict]
+            What key:value metadata tags to apply to the asset.
+        viewable_to_everyone : bool
+          If set to true, then the data asset will be shared with everyone.
+          Default is false.
+
+        Returns
+        -------
+        requests.Response
+
+        """
+        print(asset_name)
+        reg_result_response = self.co_client.register_result_as_data_asset(
+            computation_id=computation_id,
+            asset_name=asset_name,
+            mount=mount,
+            tags=tags,
+            custom_metadata=custom_metadata,
+        )
+        registered_results_response_json = reg_result_response.json()
+
+        # TODO: This step intermittently breaks. Adding extra check to help
+        #  figure out why.
+        if registered_results_response_json.get("id") is None:
+            raise KeyError(
+                f"Something went wrong registering {asset_name}. "
+                f"Response Status Code: {reg_result_response.status_code}. "
+                f"Response Message: {registered_results_response_json}"
+            )
+
+        results_data_asset_id = registered_results_response_json["id"]
+        response_res_available = self.wait_for_data_availability(
+            data_asset_id=results_data_asset_id
+        )
+
+        if response_res_available.status_code != 200:
+            raise FileNotFoundError(f"Unable to find: {results_data_asset_id}")
+
+        # Make captured results viewable to everyone
+        if viewable_to_everyone:
+            update_res_perm_response = self.co_client.update_permissions(
+                data_asset_id=results_data_asset_id, everyone="viewer"
+            )
+            print(f"Updating permissions {update_res_perm_response.status_code}")
+        return reg_result_response
+
+    def run_job(
+        self,
+        pause_interval: int = 300,
+        run_capsule_config : dict = {},
+        capture_results : bool = True,
+        capture_result_config : dict = {},
+        capture_results_asset_name : Optional[str] = None,
+        capture_results_mount : Optional[str] = None,
+    ):
+        """
+        Method to run the pipeline job
+        """
+        # 1. check data assets exist!
+        for data_asset in self.data_assets:
```

Comment:
@alejoe91 I think we should put `register_data` back in `run_job` so that there is a single entry-point for a complete workflow. Otherwise `aind-data-transfer` will have to absorb some Code Ocean logic. I think it makes more sense for `aind-data-transfer` to think about data organization and S3 and that's it.